update.ps1

```diff
@@ -65,145 +50,191 @@ function Get-ADPAccessToken {
+                                        workerID = @{
+                                            idValue = $actionContext.Data.workerId
+                                        }
                                     }
                                 }
-                            }
-                            transform    = @{
-                                worker = @{
-                                    businessCommunication = @{
-                                        email = @{
-                                            emailUri = $account.workerEmail
+                                transform    = @{
+                                    worker = @{
+                                        businessCommunication = @{
+                                            email = @{
+                                                emailUri = $actionContext.Data.workerEmail
+                                            }
                                         }
                                     }
                                 }
                             }
-                        }
-                    })
-            } | ConvertTo-Json -Depth 10
-
-            $splatParams = @{
-                Uri         = "$($config.BaseUrl)/events/hr/v1/worker.business-communication.email.change"
-                Method      = 'POST'
-                Body        = $body
-                Headers     = $headers
-                Certificate = $certificate
-                ContentType = 'application/json'
-            }
-            if (-not($dryRun -eq $true)) {
-                $responseUpdateUser = Invoke-RestMethod @splatParams
-                if ($responseUpdateUser.events[0].eventStatusCode.codeValue -eq 'submitted') {
-                    $accountReference = $responseGetUser.Workers[0].associateOID
-                    $success = $true
-                    $auditLogs.Add([PSCustomObject]@{
-                            Message = "Updated E-mail address for: $($p.DisplayName) from: [$($responseGetUser.Workers[0].businessCommunication.emails[0].emailUri)] to: [$($account.workerEmail)]"
-                            IsError = $false
                         })
+                } | ConvertTo-Json -Depth 10
+    
+                $splatParams = @{
+                    Uri         = "$($actionContext.Configuration.BaseUrl)/events/hr/v1/worker.business-communication.email.change"
+                    Method      = 'POST'
+                    Body        = $body
+                    Headers     = $headers
+                    Certificate = $certificate
+                    ContentType = 'application/json'
+                }
+                if (-not($dryRun -eq $true)) {
+                    $responseUpdateUser = Invoke-RestMethod @splatParams
+                    if ($responseUpdateUser.events[0].eventStatusCode.codeValue -eq 'submitted') {
+                        $outputContext.AccountReference = $responseGetUser.Workers[0].associateOID
```

Comment:
We can safely remove this `$outputContext.AccountReference` declaration since the associateOID probably hasn't changed. So there's no need to update the accountReference.